src/components/PaginatedTable/index.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+// import classNames from 'classnames';
+import Pagination from '@/components/Pagination';
+
+import styles from '@/components/PaginatedTable/PaginatedTable.css';
+import Table from '@/components/Table';
+export interface Props {
+  currentPage: number;
+  handlePageChange: (value: number) => void;
+  headerList: React.ReactNode[];
+  numberOfItems: number;
+  numItemsPerPage: number;
+  tableContent: React.ReactNode[][];
+  variablesClassName?: string;
+}
+
+const PaginatedTable = (props: Props) => {
+  const {
+    currentPage,
+    handlePageChange,
+    headerList,
+    numberOfItems,
+    numItemsPerPage,
+    tableContent,
+    variablesClassName
+  } = props;
+
+  const numberOfPages = Math.ceil(numberOfItems / numItemsPerPage);
+
+  const handleChange = event => {
+    if (event.target.value >= 1 && event.target.value <= numberOfPages) {
+      handlePageChange(event.target.value);
+    }
+  };
+
+  const itemsRangeBegin = (currentPage - 1) * numItemsPerPage + 1;
+  const itemsRangeEnd = () => {
+    const total = itemsRangeBegin + numItemsPerPage;
+    return total > numberOfItems ? numberOfItems : total;
+  };
+
+  const widthFirstFooterCell = headerList.length - 2;
+
+  return (
+    <Table
+      headerList={headerList}
+      tableContent={tableContent}
+      variablesClassName={variablesClassName}
+      renderFooter={() => (
+        <tfoot>
+          <tr>
+            <td className='pagination-items' style={{ lineHeight: 2 }}>
+              {itemsRangeBegin} - {itemsRangeEnd()} of {numberOfItems} items
+            </td>
+            <td className={styles['number-of-pages']} colSpan={widthFirstFooterCell}>
+              <input type='number' value={currentPage} onChange={handleChange} />
+              of {numberOfPages} pages
+            </td>
+            <td className={styles['paginated-table-action-btns']}>
+              <Pagination
+                totalPages={numberOfPages}
+                numberOfPagesToShow={0}
+                currentPage={currentPage}
+                onSelectPage={handlePageChange}
+                buttonNextLabel='Next'
+                buttonNextIcon={<span> &rarr; </span>}
+                buttonNextIconPosition='right'
+                buttonPrevLabel='Prev'
+                buttonPrevIcon={<span> &larr; </span>}
+                buttonPrevIconPosition='left'
+                buttonNextVariablesClassName={styles['next-btn-table-paginated']}
```

Comment:
@branquinhoaa is there a reason why we need two properties here? Their objectives seem to be the same: to pass in a custom CSS class that will be the same for both buttons. It seems we can simply use one property here for this. Please feel free to correct if I'm wrong.